frappe/model/db_query.py

```diff
@@ -324,10 +328,15 @@ def parse_args(self):
 					field, alias = field.split(" as ")
 				linked_fieldname, fieldname = field.split(".")
 				linked_field = frappe.get_meta(self.doctype).get_field(linked_fieldname)
+
 				linked_doctype = linked_field.options
+				linked_table = None
 				if linked_field.fieldtype == "Link":
-					self.append_link_table(linked_doctype, linked_fieldname)
-				field = f"`tab{linked_doctype}`.`{fieldname}`"
+					linked_table = self.append_link_table(linked_doctype, linked_fieldname)
+				if linked_table:
+					field = f"`{linked_table.table_alias}`.`{fieldname}`"
+				else:
+					field = f"`tab{linked_doctype}`.`{fieldname}`"
```

Comment:
I just tried to guess what this does. 

This is a bad fix as it implicitly renames fields. "Add column" feature should be handling this not db_query.


@phot0n / @Aradhya-Tripathi can you look at this?

Reply:
I'm currently tackling this, I've tried a similar approach. 
Just adding notes here that can be referenced later.
```python
if linked_table:
	 field = f"`{linked_table.table_alias}`.`{fieldname}`"
```
This doesn't work because it will try to run a permission check on the aliased table name.
https://github.com/frappe/frappe/blob/20ff30896ab540e8891f4c2a8cb670f3b4bd769b/frappe/model/db_query.py#L452